mkosi

```diff
@@ -2640,6 +2667,10 @@ def run_build_script(args, workspace, raw):
         else:
             cmdline.append("--chdir=/root")
 
+        if args.build_dir is not None:
+            cmdline.append("--setenv=BUILDDIR=/root/build")
```

Comment:
Hmm, so what would you suggest we set $BUILDDIR to if mkosi.builddir isn't set?

I mean, meson & co don't support in-tree builds, hence for them you'd probably want to se BUILDDIR=$SRCDIR/build or something like that... But automake supports in-tree builds just fine, hence then you'd want to set BUILDDIR=$SRCDIR... Hence, what would mkosi do? force things to be out-of-tree? for things to be in-tree then? I am sure we'd break half of the existing projects by either choice...

Hence I think it's best to leave this choice to the build scripts. For example in https://github.com/systemd/systemd/pull/6349 I opted for setting BUILDDIR=$SRCDIR/build if $BUILDDIR isn't set. But I am also sure for an automake project I'd want to do this differently...

Moreover, the check involving realpath isn't really pretty at all I must say...